plugins/kubernetes-backend/src/service/KubernetesFanOutHandler.ts

```diff
@@ -288,4 +285,17 @@ export class KubernetesFanOutHandler {
 
     return Promise.all([result, Promise.all(podMetrics)]);
   }
+
+  private getAuthTranslator(provider: string): KubernetesAuthTranslator {
+    if (this.authTranslators[provider]) {
+      return this.authTranslators[provider];
+    }
+
+    this.authTranslators[provider] =
+      KubernetesAuthTranslatorGenerator.getKubernetesAuthTranslatorInstance(
+        this.logger,
+        provider,
```

Comment:
I'm thinking we can keep this signature close to what we'd want for a public API, just in case. In this case that'd be `(provider, { logger })`, keeping the provider first as that's the main parameters, and then an object with dependencies so that we can easily add more of them in the future without creating an awkward long parameter list.